overrideJavaArguments

Ladvace opened the pull request.

Ladvace committed:
Basic ServerManager Css

2 colum in the servermanager

Ladvace committed:
Basic ServerList implementation

server list basic logic

Ladvace committed:
Merge remote-tracking branch 'gorilla-devs/develop'

Ladvace committed:
Using react hooks in the serverManager

Ladvace committed:
ServerManager Redux connection 

redux connection + basic button + use of react hooks

Ladvace committed:
Server Modal

Ladvace committed:
Add Server Button

Ladvace committed:
First Implementation of creation of servers

Ladvace committed:
Server Instances Creation

Ladvace committed:
Server Instances Creation fix + (useEffect clitch)

Ladvace committed:
Server Start Iplementation

Ladvace committed:
Merge remote-tracking branch 'gorilla-devs/develop'

Ladvace committed:
Kill Server Process

Ladvace committed:
Merge remote-tracking branch 'gorilla-devs/develop'

Ladvace committed:
Merge remote-tracking branch 'gorilla-devs/develop'

Ladvace committed:
Process Kill

Ladvace committed:
Merge remote-tracking branch 'gorilla-devs/develop'

Ladvace committed:
Delete Servers

Ladvace committed:
Server Settings (not finished yet)

Ladvace committed:
Server Settings Implementation (to fix)

Ladvace committed:
Server settings finished  (some improvment todo)

Ladvace committed:
Merge remote-tracking branch 'gorilla-devs/develop'

blarfoon committed:
Updated readme, Fixed ram in CLI, added game stream output

blarfoon committed:
Fixed cwd missing when starting instances

blarfoon committed:
Merge remote-tracking branch 'origin/master' into develop

blarfoon committed:
Fixed high CPU & Disk after installing an instance

blarfoon committed:
Updated default, mix and max windows sizes

blarfoon committed:
Some performance improvements on renders

blarfoon committed:
Removed unused container

blarfoon committed:
Removed aria-label from react-content-loader

blarfoon committed:
Updated login APIs. The launcher now uses directly Mojang's endpoints

blarfoon committed:
Fixed sounds not being played on older minecraft versions (down to 1.6.1)

blarfoon committed:
Added all folders to "clear instances"

blarfoon committed:
Changelog updated

blarfoon committed:
Fixed modmanager crashing when searching for a mod that doesn't exist

blarfoon committed:
Implemented instances delete confirmation

blarfoon committed:
JavaManager adapted for 32 bit systems too

blarfoon committed:
Fixed typo in javamanager

blarfoon committed:
Added default modpack name when installing

blarfoon committed:
Fixed modal size when app is fullscreen

blarfoon committed:
Added modal blur effect (PROPOSAL)

blarfoon committed:
Updated npm packages

blarfoon committed:
Added validation of instances names

blarfoon committed:
Allowed experimental features

Ladvace committed:
Merge remote-tracking branch 'gorilla-devs/develop'

blarfoon committed:
Added padding to contextmenu header

blarfoon committed:
Added new user page

Ladvace committed:
Merge remote-tracking branch 'gorilla-devs/develop'

Ladvace committed:
serverCommand implementation

Ladvace committed:
Merge remote-tracking branch 'gorilla-devs/develop' into develop

Ladvace committed:
Merge remote-tracking branch 'gorilla-devs/develop'

Ladvace committed:
Revert "Merge remote-tracking branch 'gorilla-devs/develop'"

This reverts commit a4c0a13b1219170bf17a6ab14529550dcd2f6ce6.

Ladvace committed:
Merge remote-tracking branch 'gorilla-devs/develop'

Ladvace committed:
commandButton css

Ladvace committed:
random

Ladvace committed:
Commands ServerManager fix

Ladvace committed:
Merge remote-tracking branch 'gorilla-devs/develop'

Ladvace committed:
runCommand (server) implementation

Ladvace committed:
Merge remote-tracking branch 'gorilla-devs/develop'

Ladvace committed:
little text fix

Ladvace committed:
Show public ip

Ladvace committed:
Merge remote-tracking branch 'gorilla-devs/develop'

Ladvace committed:
server command creation implementation

Ladvace committed:
Merge branch 'master' into gorilla-devs/develop

Ladvace committed:
remove button (server command)

button group

Ladvace committed:
finished remove button

Ladvace committed:
little fix

Ladvace committed:
little fix

Ladvace committed:
message box invalid input

Ladvace committed:
added default server commands

Ladvace committed:
Fix--Re-added Create Server Button (wrongly removed)

Ladvace committed:
ServerCommands.json File (not finished yet)

Ladvace committed:
Merge remote-tracking branch 'gorilla-devs/develop'

Ladvace committed:
Added Name to commandManager

Ladvace committed:
Merge remote-tracking branch 'gorilla-devs/develop'

Ladvace committed:
Added name to serverSettings manager

Ladvace committed:
Commented usless code (maybe)

Ladvace committed:
Merge remote-tracking branch 'gorilla-devs/master'

Ladvace committed:
some few changes and (i hope) fix

Ladvace committed:
remove a (unknown) file

Ladvace committed:
Merge remote-tracking branch 'gorilla-devs/master'

Ladvace committed:
Working server command file

Ladvace committed:
Fixed server command

Ladvace committed:
clean the input after the creation of the command

Ladvace committed:
css button styles fix

Ladvace committed:
missing server properties file error message box

Ladvace committed:
max-height server box

Ladvace committed:
Server Fixed (server command problems)

Ladvace committed:
Merge remote-tracking branch 'gorilla-devs/develop'

Ladvace committed:
style fix

Ladvace committed:
Font styles

Ladvace committed:
css style fix

Ladvace committed:
css fix

Ladvace committed:
Merge remote-tracking branch 'gorilla-devs/develop'

Ladvace committed:
ASYNC USEFFECT FIXED

Ladvace committed:
Run serverCommand FIXED (I HOPE)

Ladvace committed:
nogui added

Ladvace committed:
HorizonalMenu active selection

Ladvace committed:
Merge remote-tracking branch 'gorilla-devs/master'

Ladvace committed:
Server Improvement (important commit)

Ladvace committed:
Merge remote-tracking branch 'gorilla-devs/develop'

Ladvace committed:
servers icon

Ladvace committed:
server fix

Ladvace committed:
close event

Ladvace committed:
close event Almost finished

Ladvace committed:
Merge remote-tracking branch 'gorilla-devs/master'

Ladvace committed:
close event Improvement

Ladvace committed:
close all servers on close event

Ladvace committed:
removed some useless commented code

Ladvace committed:
removed commented code

Ladvace committed:
serverManager styling (align center)

Ladvace committed:
serverManager Icons

Ladvace committed:
servers style

Ladvace committed:
glitch fix

Ladvace committed:
Merge remote-tracking branch 'gorilla-devs/develop'

Ladvace committed:
delete server modal

Ladvace committed:
add and play group button servers commands

Ladvace committed:
runCustomCommand

Ladvace committed:
start refactoring

Ladvace committed:
ServerCommandsModal + others improvements

Ladvace committed:
serverCommandsModal title

Ladvace committed:
Server Commands Modal button style

Ladvace committed:
some fix

Ladvace committed:
deleted commented useless code

Ladvace committed:
fix

Ladvace committed:
UseEffect /list players

Ladvace committed:
Merge remote-tracking branch 'gorilla-devs/develop' into develop

Ladvace committed:
Override java arguments form

Ladvace committed:
JavaMemorySlider for InstancesManager

Ladvace committed:
instancesManager made functional

Ladvace committed:
scroll settings fixed

Ladvace committed:
first implementation javaslider instancesManager

Ladvace committed:
java args in the input

Ladvace committed:
switch javaMemory slider css style

Ladvace committed:
switch css style fix

Ladvace committed:
switch css margin

Ladvace committed:
set overrideArgs in instance config.json for the overrideJavaArguments + some fix

Ladvace committed:
switchState checker (overrideJavaArguments instances)

Ladvace committed:
java overrideMemory

 (separation between override instance memory and global java memory)

Ladvace committed:
java overrideMemory (separation between override instance memory and global java memory)

Ladvace committed:
Merge remote-tracking branch 'origin/develop' into develop

Ladvace committed:
overrideJavaArguments fixed

Ladvace committed:
overrideJavaArguments finished fix

Ladvace committed:
Default java args instances (overrideJavaArgs)

Ladvace committed:
Merge remote-tracking branch 'gorilla-devs/develop' into develop

Ladvace committed:
code style fix

Ladvace committed:
Merge branch 'develop'

Ladvace committed:
merge fix

Ladvace committed:
merge fix

hidden server tab from horizontal menu

Ladvace committed:
Merge remote-tracking branch 'origin/master'

Ladvace committed:
css code style fix

Ladvace committed:
style fix

Ladvace committed:
css changes

Ladvace committed:
fix

Ladvace committed:
build fix

some fix

Ladvace committed:
fix

Ladvace committed:
serverCommand tab commented

Ladvace committed:
switch fix

Ladvace committed:
css overrideJavaArguments

Ladvace committed:
not working yet hiding switch

Ladvace committed:
javaMemorySlider fix

Ladvace committed:
working override java memory

Ladvace committed:
working override java args

Ladvace committed:
code and css fix

blarfoon commented on app/actions/downloadManager.js:
```diff
@@ -391,7 +392,9 @@ export function downloadPack(pack) {
-        overrideFiles: overrideFilesList
+        overrideFiles: overrideFilesList,
+        overrideArgs: DEFAULT_ARGS,
+        overrideMemory: 3096
```
By default none of those should be present. Only when an override is set it is added. If no override is set, it is simply not present and the global is used instead

blarfoon commented on app/actions/instancesManager.js:
```diff
@@ -41,10 +41,18 @@ export function startInstance(instanceName) {
       instanceName,
       auth,
-      settings.java.memory,
+      config.overrideMemory === ''
```
it should be`config.overrideMemory === undefined` since when no override is set, it is simply not present

blarfoon commented on app/actions/serverManager.js:
```diff
@@ -0,0 +1,84 @@
+import { message } from 'antd';
```
Why is serverManager in this PR?

blarfoon commented on app/components/Common/WindowNavigation/components/HorizontalMenu/HorizontalMenu.js:
```diff
@@ -53,18 +53,21 @@ export default props => {
+          <Link
+            to="/serverManager"
+            draggable="false"
+            className={styles.a}
```
Why is this in this PR?

blarfoon commented on app/components/ServerCreatorModal/ServerCreatorModal.js:
```diff
@@ -0,0 +1,177 @@
+import Modal from '../Common/Modal/Modal';
+import axios from 'axios';
+import { downloadFile } from '../../utils/downloader';
+import styles from './ServerCreatorModal.scss';
```
Why is this file in this PR?

blarfoon commented on app/components/ServerCreatorModal/ServerCreatorModal.scss:
```diff
@@ -0,0 +1,16 @@
+  width: 100%;
+  padding: 20px;
+}
+
```
Why is this file in this PR?

blarfoon commented on app/components/ServerManager/ServerManager.js:
```diff
@@ -1,27 +1,185 @@
+import log from 'electron-log';
+import { exec } from 'child_process';
+import path from 'path';
+import psTree from 'ps-tree';
```
Why is this file in this PR?

blarfoon commented on app/components/ServerManager/ServerManager.scss:
```diff
@@ -1,35 +1,160 @@
-  grid-gap: 3%
+}
+
+.serverSettings{
```
Why is this file in this PR?

blarfoon commented on app/components/ServerManager/command.js:
```diff
@@ -0,0 +1,56 @@
+}
+
+
+function Command(props) {
```
Why is this file in this PR?

blarfoon commented on app/components/ServerManager/serverCommand.js:
```diff
@@ -0,0 +1,219 @@
+import { downloadFile } from '../../utils/downloader';
+import { SERVERS_PATH } from '../../constants';
+import ButtonGroup from 'antd/lib/button/button-group';
+import Command from './command';
```
Why is this file in this PR?

blarfoon commented on app/components/serverCommandsModal/serverCommandsModal.js:
```diff
@@ -0,0 +1,100 @@
+import React, { useState, useEffect } from 'react';
+import { connect } from 'react-redux';
+import { message, Button } from 'antd';
+import path from 'path';
```
Why is this file in this PR?

blarfoon commented on app/components/serverCommandsModal/serverCommandsModal.scss:
```diff
@@ -0,0 +1,35 @@
+  // margin: 1%;
+  align-items: center;
+  justify-content: center;
+  overflow-y: auto;
```
Why is this file in this PR?

blarfoon commented on app/reducers/serverManager.js:
```diff
@@ -0,0 +1,32 @@
+  switch (action.type) {
+    case `${START_SERVER}`:
+      return {
+        ...state,
```
Why is this file in this PR?

blarfoon commented on package.json:
```diff
@@ -184,6 +184,7 @@
     "@fortawesome/free-regular-svg-icons": "^5.7.2",
     "@fortawesome/free-solid-svg-icons": "^5.7.2",
     "@fortawesome/react-fontawesome": "^0.1.4",
+    "@rauschma/stringio": "^1.4.0",
```
Remove this line since is not needed

Ladvace committed:
removed serverManager Files

Ladvace committed:
removed servers files

Ladvace committed:
Merge remote-tracking branch 'origin/master'

Ladvace committed:
removed stringio dependency

Ladvace committed:
override java args & memory system modified

now property are not added to the config.json when it's created but only if the override is active

Ladvace committed:
fix

Ladvace committed:
some java ovveride args fix

blarfoon commented on app/actions/downloadManager.js:
```diff
@@ -28,6 +28,7 @@ import { downloadMod, getModsList, createDoNotTouchFile } from '../utils/mods';
 import { arraify } from '../utils/strings';
 import { copyAssetsToLegacy, copyAssetsToResources } from '../utils/assets';
 import { getAddonFile, getAddon } from '../utils/cursemeta';
+import { DEFAULT_ARGS } from '../constants';
```
Why importing an unused variable?

blarfoon commented on app/components/InstanceManagerModal/Settings/JavaMemorySlider.js:
```diff
@@ -0,0 +1,146 @@
+import { toggleOverrideJavaArguments } from '../../../actions/settings';
+import { PACKS_PATH } from '../../../constants';
+
+function javaMemorySlider(props) {
```
components names should be pascal-case

blarfoon commented on app/components/InstanceManagerModal/Settings/JavaMemorySlider.js:
```diff
@@ -0,0 +1,146 @@
+          path.join(PACKS_PATH, props.instanceName, 'config.json')
+        )
+      );
+      if (config.overrideMemory === undefined && e === true) {
```
e === true is equal to e

blarfoon commented on app/components/InstanceManagerModal/Settings/JavaMemorySlider.js:
```diff
@@ -0,0 +1,146 @@
+          modifiedConfig
+        );
+        setSwitchState(true);
+      } else if (config.overrideMemory !== undefined && e === false) {
```
e === true is equal to e

blarfoon commented on app/components/InstanceManagerModal/Settings/Settings.js:
```diff
@@ -1,172 +1,341 @@
+        path.join(PACKS_PATH, props.instance, 'config.json')
+      )
+    );
+    config.overrideArgs = DEFAULT_ARGS;
```
Useless assignment

blarfoon commented on app/components/InstanceManagerModal/Settings/Settings.js:
```diff
@@ -1,172 +1,341 @@
+      )
+    );
+    config.overrideArgs = DEFAULT_ARGS;
+    const modifiedConfig = JSON.stringify(config);
```
use
`JSON.stringify({
...config,
overrideArgs: DEFAULT_ARGS
})`

blarfoon commented on app/components/InstanceManagerModal/Settings/Settings.js:
```diff
@@ -1,172 +1,341 @@
+          path.join(PACKS_PATH, props.instance, 'config.json')
+        )
+      );
+      config.overrideArgs = overrideArgs;
```
Try not to mutate objects when possible 🥇

blarfoon commented on app/components/InstanceManagerModal/Settings/Settings.js:
```diff
@@ -1,172 +1,341 @@
+        )
+      );
+      config.overrideArgs = overrideArgs;
+      const modifiedConfig = JSON.stringify(config);
```
JSON.stringify({ ...config, overrideArgs: DEFAULT_ARGS })

blarfoon commented on app/components/InstanceManagerModal/Settings/Settings.js:
```diff
@@ -1,172 +1,341 @@
+    } else message.error('Enter Valid Arguments');
+  };
+
+  async function configManagment() {
```
configManagement please

blarfoon commented on app/components/InstanceManagerModal/Settings/Settings.js:
```diff
@@ -1,172 +1,341 @@
+        path.join(PACKS_PATH, props.instance, 'config.json')
+      )
+    );
+    if (configFile.overrideArgs !== undefined) {
```
You can safely remove !== undefined

blarfoon commented on app/components/InstanceManagerModal/Settings/Settings.js:
```diff
@@ -1,172 +1,341 @@
-    watcher.close();
-  };
+  useEffect(() => {
+    configManagment();
```
management <------------

blarfoon commented on app/components/InstanceManagerModal/Settings/Settings.js:
```diff
@@ -1,172 +1,341 @@
+  };
+
+  // Set the changed java arguments
+  const submit = async () => {
```
Use a more specific name for this function

blarfoon commented on app/components/InstanceManagerModal/Settings/Settings.js:
```diff
@@ -1,172 +1,341 @@
+  };
+
+  // Reset the global arguments to the default one
+  const reset = async () => {
```
Use a more specific name for this function

blarfoon commented on app/components/InstanceManagerModal/Settings/Settings.js:
```diff
@@ -1,172 +1,341 @@
+          path.join(PACKS_PATH, props.instance, 'config.json')
+        )
+      );
+      if (config.overrideArgs === undefined && e === true) {
```
I think you can remove the last ===

blarfoon commented on app/components/InstanceManagerModal/Settings/Settings.js:
```diff
@@ -1,172 +1,341 @@
+        )
+      );
+      if (config.overrideArgs === undefined && e === true) {
+        config.overrideArgs = props.overrideJavaArgs;
```
Same as above

blarfoon commented on app/components/InstanceManagerModal/Settings/Settings.js:
```diff
@@ -1,172 +1,341 @@
+      );
+      if (config.overrideArgs === undefined && e === true) {
+        config.overrideArgs = props.overrideJavaArgs;
+        const modifiedConfig = JSON.stringify(config);
```
Same as above

blarfoon commented on app/components/InstanceManagerModal/Settings/Settings.js:
```diff
@@ -1,172 +1,341 @@
+        );
+        setOverrideArgsInput(props.overrideJavaArgs);
+        setSwitchState(true);
+      } else if (config.overrideArgs !== undefined && e === false) {
```
You can remove both ===

blarfoon commented on app/components/InstanceManagerModal/Settings/Settings.js:
```diff
@@ -1,172 +1,341 @@
+        setOverrideArgsInput(props.overrideJavaArgs);
+        setSwitchState(true);
+      } else if (config.overrideArgs !== undefined && e === false) {
+        config.overrideArgs = undefined;
```
Same as above

blarfoon commented on app/components/InstanceManagerModal/Settings/Settings.js:
```diff
@@ -1,172 +1,341 @@
+        setSwitchState(true);
+      } else if (config.overrideArgs !== undefined && e === false) {
+        config.overrideArgs = undefined;
+        const modifiedConfig = JSON.stringify(config);
```
Same as above

blarfoon commented on app/components/InstanceManagerModal/Settings/Settings.js:
```diff
@@ -1,172 +1,341 @@
+                  checked={switchState}
+                />
+              </div>
+              {switchState ? javaArgInput : <div />}
```
Return null instead of an empty div. Decreases overall complexity

blarfoon commented on app/reducers/settings.js:
```diff
@@ -15,7 +17,9 @@ const initialState = {
-    javaArgs: DEFAULT_ARGS
+    overrideMemory: 3096,
+    javaArgs: DEFAULT_ARGS,
+    overrideJavaArgs: DEFAULT_ARGS
```
Imho this is completely useless and you can get rid of it

blarfoon commented on app/utils/MCLaunchCommand.js:
```diff
@@ -58,12 +58,18 @@ const getStartCommand = async (packName, userData, ram, javaArguments) => {
+    await promisify(fs.readFile)(path.join(PACKS_PATH, packName, 'config.json'))
+  );
+
+  // const ramMemory = config.overrieMemory === '' ? ram : config.overrieMemory;
```
???????

blarfoon commented on app/utils/MCLaunchCommand.js:
```diff
@@ -58,12 +58,18 @@ const getStartCommand = async (packName, userData, ram, javaArguments) => {
-    '{_RAM_}',
-    ram
-  )} ${dosName} -Djava.library.path="${path.join(
+"${javaPath}" ${(config.overrideArgs === undefined
```
You can use `config.overrideArgs ? config.overrideArgs : javaArguments`

blarfoon commented on yarn.lock:
```diff
@@ -1188,13 +1188,27 @@
     url-template "^2.0.8"
 
-"@sinonjs/commons@^1", "@sinonjs/commons@^1.0.2", "@sinonjs/commons@^1.3.1":
+"@rauschma/stringio@^1.4.0":
```
I would appreciate if you could remove this file from the PR. Thanks

blarfoon reviewed (changes requested):
See comments

Ladvace committed:
code style fix

blarfoon commented on app/components/InstanceManagerModal/Settings/JavaMemorySlider.js:
```diff
@@ -0,0 +1,145 @@
+        path.join(PACKS_PATH, props.instanceName, 'config.json')
+      )
+    );
+    if (configFile.overrideMemory !== undefined) {
```
This can be written as `if(config.overrideMemory)`

blarfoon commented on app/actions/instancesManager.js:
```diff
@@ -41,10 +41,18 @@ export function startInstance(instanceName) {
       instanceName,
       auth,
-      settings.java.memory,
+      config.overrideMemory === undefined
```
This can be written as
`config.overrideMemory ? config.overrideMemory : settings.java.memory`

blarfoon commented on app/components/InstanceManagerModal/Settings/JavaMemorySlider.js:
```diff
@@ -0,0 +1,145 @@
+        )
+      );
+      if (config.overrideMemory === undefined && e) {
+        config.overrideMemory = props.overrideMemory;
```
Try not to mutate objects

blarfoon commented on app/components/InstanceManagerModal/Settings/JavaMemorySlider.js:
```diff
@@ -0,0 +1,145 @@
+        );
+        setSwitchState(true);
+      } else if (config.overrideMemory !== undefined && !e) {
+        delete config.overrideMemory;
```
Try not to mutate objects

blarfoon commented on app/components/InstanceManagerModal/Settings/JavaMemorySlider.js:
```diff
@@ -0,0 +1,145 @@
+          checked={switchState}
+        />
+      </div>
+      {switchState ? memorySlider : <div />}
```
Don't return empty divs, return null instead

blarfoon commented on app/components/InstanceManagerModal/Settings/Settings.js:
```diff
@@ -1,172 +1,337 @@
+        )
+      );
+      if (config.overrideArgs === undefined && e) {
+        config.overrideArgs = props.overrideJavaArgs;
```
Try not to mutate objects

blarfoon commented on app/components/InstanceManagerModal/Settings/Settings.js:
```diff
@@ -1,172 +1,337 @@
+        setOverrideArgsInput(props.overrideJavaArgs);
+        setSwitchState(true);
+      } else if (config.overrideArgs && !e) {
+        config.overrideArgs = undefined;
```
Try not to mutate objects

blarfoon commented on app/reducers/settings.js:
```diff
@@ -15,6 +17,7 @@ const initialState = {
     autodetected: true,
     path: null,
     memory: 3096,
+    overrideMemory: 3096,
```
If I change the override memory for one instance, it changes it for all the instances

blarfoon reviewed (changes requested):
See comments and fix bugs

Ladvace committed:
some fix

Ladvace committed:
Revert "partial working"

This reverts commit dd2b1186408097272289f4282f9570129d687af4.

Ladvace committed:
partial working

Ladvace committed:
some changes

Ladvace committed:
Merge remote-tracking branch 'origin/master'

Ladvace committed:
fix

Ladvace committed:
fix

Ladvace committed:
console.logs removed

blarfoon commented on app/actions/instancesManager.js:
```diff
@@ -41,10 +41,16 @@ export function startInstance(instanceName) {
       instanceName,
       auth,
-      settings.java.memory,
+      config.overrideMemory ? config.overrideMemory : settings.java.memory,
```
Write `config.overrideMemory || settings.java.memory` instead

blarfoon commented on app/reducers/settings.js:
```diff
@@ -5,7 +5,7 @@ import {
   SET_THEME,
   RESET_THEME,
-  SET_GLOBAL_JAVA_ARGUMENTS
+  SET_GLOBAL_JAVA_ARGUMENTS,
```
What is this used for?

blarfoon commented on app/routes.js:
```diff
@@ -305,4 +300,4 @@ function mapDispatchToProps(dispatch) {
 export default connect(
   mapStateToProps,
   mapDispatchToProps
-)(RouteDef);
```
Don't remove newline at the end of the file

blarfoon commented on app/utils/MCLaunchCommand.js:
```diff
@@ -58,12 +58,16 @@ const getStartCommand = async (packName, userData, ram, javaArguments) => {
-    '{_RAM_}',
-    ram
-  )} ${dosName} -Djava.library.path="${path.join(
+"${javaPath}" ${(config.overrideArgs
```
Can write `config.overrideArgs || javaArguments` instead

blarfoon commented on yarn.lock:
```diff
@@ -1188,13 +1188,20 @@
     url-template "^2.0.8"
 
-"@sinonjs/commons@^1", "@sinonjs/commons@^1.0.2", "@sinonjs/commons@^1.3.1":
+"@sinonjs/commons@^1", "@sinonjs/commons@^1.3.1":
```
Don't include yarn.lock to the PR
